server/src/test/java/org/elasticsearch/search/aggregations/pipeline/BucketSortIT.java

```diff
@@ -231,6 +252,29 @@ public void testSortTermsOnSubAggregation() {
         }
     }
 
+    public void testSortTermsOnSubAggregationPreservesOrderOnEquals() {
+        SearchResponse response = client().prepareSearch(INDEX)
+            .setSize(0)
+            .addAggregation(terms("foos").field(TERM_FIELD)
+                .subAggregation(bucketSort("bucketSort", Arrays.asList(new FieldSortBuilder("_key"))))
```

Comment:
It might be nice to name the two `bucket_sort` aggs differently for readability.